drivers/storage/portworx/preflight.go

```diff
@@ -180,8 +185,36 @@ func (u *preFlightPortworx) RunPreFlight() error {
 	   	u.cluster.Annotations[pxutil.AnnotationMiscArgs] = strings.TrimSpace(miscArgs)
 	*/
 
-	// Add --pre-flight check for DMthin
-	preflightDS.Spec.Template.Spec.Containers[0].Args = append([]string{"--pre-flight", "-T", "dmthin"},
+	checkArgs := func(args []string) {
+		for i, arg := range args {
+			if arg == "-T" {
+				if dmthinRegex.Match([]byte(args[i+1])) {
+					u.hardFail = true
+				}
+			}
+		}
+	}
+
+	// Check for pre-existing DMthin in misc-args
+	if mArgs, err := pxutil.MiscArgs(u.cluster); err == nil {
+		checkArgs(mArgs)
+	} else {
+		logrus.Warnf("runPreFlight: error parsing misc args, skipping existing PX-StoreV2 param check: %v", err)
+	}
```

Comment:
We are already checking the pod args below. Why do we need to check the cluster's misc args again?

Reply:
Args can be added in the spec correct.  So we should check them both..